src/Connector.SqlServer/Connector/SqlClient.cs

```diff
@@ -22,17 +23,62 @@ public string BuildConnectionString(IReadOnlyDictionary<string, object> config)
                 DataSource = (string)config[SqlServerConstants.KeyName.Host],
                 InitialCatalog = (string)config[SqlServerConstants.KeyName.DatabaseName],
                 Pooling = true,
-                MaxPoolSize = 200
             };
 
-            if (config.TryGetValue(SqlServerConstants.KeyName.PortNumber, out var portEntry) && int.TryParse(portEntry.ToString(), out var port))
+            // Configure port
+            {
+                var port = _defaultPort;
+                if (config.TryGetValue(SqlServerConstants.KeyName.PortNumber, out var portEntry) &&
+                    !string.IsNullOrEmpty(portEntry.ToString()) &&
+                    int.TryParse(portEntry.ToString(), out var parsedPort))
+                {
+                    port = parsedPort;
+                }
+
                 connectionStringBuilder.DataSource = $"{connectionStringBuilder.DataSource},{port}";
-            else
-                connectionStringBuilder.DataSource = $"{connectionStringBuilder.DataSource},{_defaultPort}";
+            }
+
+            // Configure connection pool size
+            {
+                var connectionPoolSize = _defaultConnectionPoolSize;
+                if (config.TryGetValue(SqlServerConstants.KeyName.ConnectionPoolSize, out var connectionPoolSizeEntry) &&
+                    !string.IsNullOrEmpty(connectionPoolSizeEntry.ToString()) &&
+                    int.TryParse(connectionPoolSizeEntry.ToString(), out var parsedConnectionPoolSize))
+                {
+                    connectionPoolSize = parsedConnectionPoolSize;
+                }
+
+                connectionStringBuilder.MaxPoolSize = connectionPoolSize;
+            }
+
 
             return connectionStringBuilder.ToString();
         }
 
+        public bool VerifyConnectionProperties(IReadOnlyDictionary<string, object> config, out ConnectionConfigurationError configurationError)
+        {
+            if (config.TryGetValue(SqlServerConstants.KeyName.PortNumber, out var portEntry) && !string.IsNullOrEmpty(portEntry.ToString()))
+            {
+                if (!int.TryParse(portEntry.ToString(), out _))
+                {
+                    configurationError = new ConnectionConfigurationError("Port number was set, but could not be read as a number");
+                    return false;
+                }
+            }
+
+            if (config.TryGetValue(SqlServerConstants.KeyName.ConnectionPoolSize, out var connectionPoolSizeEntry) && !string.IsNullOrEmpty(connectionPoolSizeEntry.ToString()))
+            {
+                if (!int.TryParse(connectionPoolSizeEntry.ToString(), out _))
+                {
+                    configurationError = new ConnectionConfigurationError("Connection pool size was set, but could not be read as a number");
+                    return false;
+                }
+            }
```

Comment:
Add validation of max / min size.

Reply:
Added validation that connection pool should not be smaller than 1, and not be bigger than 32767, since that is the max value allowed by SQL server. Added comment to explain this.

Reply:
Changed to 200